presto-memory/src/test/java/io/prestosql/plugin/memory/TestMemorySmoke.java

```diff
@@ -124,18 +154,50 @@ public void testJoinDynamicFilteringSingleValue()
                 .setSystemProperty(JOIN_DISTRIBUTION_TYPE, FeaturesConfig.JoinDistributionType.BROADCAST.name())
                 .build();
         DistributedQueryRunner runner = (DistributedQueryRunner) getQueryRunner();
-        ResultWithQueryId<MaterializedResult> result = runner.executeWithQueryId(session, "SELECT * FROM lineitem JOIN orders " +
-                "ON lineitem.orderkey = orders.orderkey AND orders.comment = 'nstructions sleep furiously among '");
+
+        // Join lineitem with a single row of orders
+        ResultWithQueryId<MaterializedResult> result = runner.executeWithQueryId(
+                session,
+                "SELECT * FROM lineitem JOIN orders ON lineitem.orderkey = orders.orderkey AND orders.comment = 'nstructions sleep furiously among '");
         assertEquals(result.getResult().getRowCount(), 6);
+        assertEquals(getOperatorRowsRead(runner, result.getQueryId()), ImmutableSet.of(6L, ORDERS_COUNT));
+
+        // Join lineitem with a single row of part
+        result = runner.executeWithQueryId(
+                session,
+                "SELECT l.comment FROM  lineitem l, part p WHERE p.partkey = l.partkey AND p.comment = 'onic deposits'");
+        assertEquals(result.getResult().getRowCount(), 39);
+        assertEquals(getOperatorRowsRead(runner, result.getQueryId()), ImmutableSet.of(39L, PART_COUNT));
+    }
 
-        // Probe-side is dynamically filtered:
-        QueryStats stats = runner.getCoordinator().getQueryManager().getFullQueryInfo(result.getQueryId()).getQueryStats();
-        Set<Long> rowsRead = stats.getOperatorSummaries()
+    @Test
+    public void testJoinDynamicFilteringBlockProbeSide()
+    {
+        Session session = Session.builder(getSession())
+                .setSystemProperty(ENABLE_DYNAMIC_FILTERING, "true")
```

Comment:
`.setSystemProperty(ENABLE_DYNAMIC_FILTERING, "true")` is not needed, so you can remove it from tests (as a separate commit)